0xProto.glyphs

```diff
@@ -481,6 +492,29 @@ code = "# lookupflag IgnoreMarks;
 ";
 disabled = 1;
 tag = liga;
+},
+{
+automatic = 1;
+code = "lookup ccmp_Other_1 {
+	@CombiningTopAccents = [acutecomb caroncomb circumflexcomb dieresiscomb dotaccentcomb gravecomb ringcomb tildecomb];
+	lookupflag UseMarkFilteringSet @CombiningTopAccents;
+	sub i' @CombiningTopAccents by idotless;
+	sub j' @CombiningTopAccents by jdotless;
+} ccmp_Other_1;
+
+lookup ccmp_Other_2 {
+	@Markscomb = [dieresiscomb dotaccentcomb gravecomb acutecomb circumflexcomb caroncomb ringcomb tildecomb cedillacomb];
+	@MarkscombCase = [dieresiscomb.case dotaccentcomb.case gravecomb.case acutecomb.case circumflexcomb.case caroncomb.case ringcomb.case tildecomb.case cedillacomb.case];
+	sub @Markscomb @Markscomb' by @MarkscombCase;
+	sub @Uppercase @Markscomb' by @MarkscombCase;
+} ccmp_Other_2;
+
+lookup ccmp_Other_3 {
+	sub @Markscomb' @MarkscombCase by @MarkscombCase;
+	sub @MarkscombCase @Markscomb' by @MarkscombCase;
+} ccmp_Other_3;
+";
+tag = ccmp;
```

Comment:
The code for this feature appears to be auto-generated by Glyphs.app, did you press the "Update" button in the lower left corner?

<img width="1295" alt="image" src="https://github.com/0xType/0xProto/assets/1401513/e4a85231-5406-47d4-9d7d-6f77caaf6cea">

I just pressed that button and it updated to the following:

<img width="1297" alt="image" src="https://github.com/0xType/0xProto/assets/1401513/5aa061ca-dcf5-4e7f-8f5c-6b77ab8cb020">

I understand that there is no change in what ccmp is doing, but I am curious as to why the code is changing.

Reply:
Glyphs.app updated the way it generates this ccmp feature code. Functionally they aim to do the same but the old code wasn’t complete.

This really will only affect your font if you add more characters, in particular marks below, as the old code only handled the context with one mark below and one mark above, whereas the new code handles any context with at least on mark above.

Reply:
Sorry, I could not see how you generated this code...

> Functionally they aim to do the same but the old code wasn’t complete.

What does "old code" here refer to? Is it 1 or 2 of the following?

### 1: auto generated on my local env

```
lookup ccmp_Other_1 {
	@CombiningTopAccents = [acutecomb caroncomb circumflexcomb dieresiscomb dotaccentcomb gravecomb ringcomb tildecomb];
	@CombiningNonTopAccents = [cedillacomb];
	sub [i j]' @CombiningTopAccents by [idotless jdotless];
	sub [i j]' @CombiningNonTopAccents @CombiningTopAccents by [idotless jdotless];
	@Markscomb = [dieresiscomb dotaccentcomb gravecomb acutecomb circumflexcomb caroncomb ringcomb tildecomb cedillacomb];
	@MarkscombCase = [dieresiscomb.case dotaccentcomb.case gravecomb.case acutecomb.case circumflexcomb.case caroncomb.case ringcomb.case tildecomb.case cedillacomb.case];
	sub @Markscomb @Markscomb' by @MarkscombCase;
	sub @Uppercase @Markscomb' by @MarkscombCase;
} ccmp_Other_1;

lookup ccmp_Other_2 {
	sub @Markscomb' @MarkscombCase by @MarkscombCase;
	sub @MarkscombCase @Markscomb' by @MarkscombCase;
} ccmp_Other_2;
```

### 2: code in this PR

```
lookup ccmp_Other_1 {
	@CombiningTopAccents = [acutecomb caroncomb circumflexcomb dieresiscomb dotaccentcomb gravecomb ringcomb tildecomb];
	lookupflag UseMarkFilteringSet @CombiningTopAccents;
	sub i' @CombiningTopAccents by idotless;
	sub j' @CombiningTopAccents by jdotless;
} ccmp_Other_1;

lookup ccmp_Other_2 {
	@Markscomb = [dieresiscomb dotaccentcomb gravecomb acutecomb circumflexcomb caroncomb ringcomb tildecomb cedillacomb];
	@MarkscombCase = [dieresiscomb.case dotaccentcomb.case gravecomb.case acutecomb.case circumflexcomb.case caroncomb.case ringcomb.case tildecomb.case cedillacomb.case];
	sub @Markscomb @Markscomb' by @MarkscombCase;
	sub @Uppercase @Markscomb' by @MarkscombCase;
} ccmp_Other_2;

lookup ccmp_Other_3 {
	sub @Markscomb' @MarkscombCase by @MarkscombCase;
	sub @MarkscombCase @Markscomb' by @MarkscombCase;
} ccmp_Other_3;
```

Reply:
Glyphs.app generates different automatic ccmp code depending on what version you have installed. I guess the automatic ccmp code I have is only generated in beta versions.

Reply:
Got it, I totally get what you mean!
I've upgraded to the latest cutting edge version of Glyphs.app (3.2), and it's now auto-generating the same code as you did.